atlas/schema.hcl

```diff
@@ -1,11 +1,476 @@
+  column "organization_id" {
+    null = false
+    type = bigint
+    identity {
+      generated = ALWAYS
+      start = 0
+      increment = 1
+    }
+  }
+ column "title" {
+    null = false
+    type = text
+    comment = "name of business"
+  }
+  column "income_type" {
+    null = false
+    type = text
+  }
+  column "industry_code" {
+    null = false
+    type = text
+  }
+  column "main_product" {
+    null = true
+    type = text
+    comment = "main product/service sold by the business"
+  }
+  column "partnership" {
+    null = false
+    type = bool
+    default = false
+  }
+  column "address" {
+    null = true
+    type = text
+    comment = "business physical address if applicable"
+  }
+  column "create_time" {
+    null = false
+    type = timestamptz
+  }
+  primary_key {
+    columns = [column.organization_id]
+  }
+}
+table "project_v1" {
+  schema = schema.public
+  column "user_id" {
+    null = false
+    type = bigint
+  }
+  column "project_id" {
+    null = false
+    type = bigint
+    identity {
+      generated = ALWAYS
+      start = 0
+      increment = 1
+    }
+  }
+  primary_key {
+    columns = [column.user_id, column.project_id]
+  }
+  foreign_key "user_id" {
+    columns     = [column.user_id]
+    ref_columns = [table.user_v1.column.user_id]
+    on_update   = CASCADE
+    on_delete   = CASCADE
+  }
+  index "project_id_idx" {
+    unique  = true
+    columns = [column.project_id]
+  }
+}
+table "scan_inference_v1" {
+  schema = schema.public
+  column "scan_id" {
+    null = false
+    type = bigint
+  }
+  column "scan_inference_id" {
+    null = false
+    type = bigint
+    identity {
+      generated = ALWAYS
+      start = 0
+      increment = 1
+    }
+  }
+  column "inference_result" {
+    null = false
+    type = bytea
+    comment = "JSON of inference output"
   }
   primary_key {
-    columns = [column.id]
+    columns = [column.scan_id, column.scan_inference_id]
+  }
+  foreign_key "scan_id" {
+    columns     = [column.scan_id]
+    ref_columns = [table.scan_v1.column.scan_id]
+    on_update   = CASCADE
+    on_delete   = CASCADE
+  }
+}
+table "scan_v1" {
+  schema = schema.public
+  column "project_id" {
+    null = false
+    type = bigint
+  }
+  column "scan_id" {
+    null = false
+    type = bigint
+    identity {
+      generated = ALWAYS
+      start = 0
+      increment = 1
+    }
+  }
+  column "image" {
+    null = false
+    type = bytea
+  }
+  column "create_time" {
+    null = false
+    type = timestamptz
+  }
+  primary_key {
+    columns = [column.project_id, column.scan_id]
+  }
+  foreign_key "project_id" {
+    columns     = [column.project_id]
+    ref_columns = [table.project_v1.column.project_id]
+    on_update   = CASCADE
+    on_delete   = CASCADE
+  }
+  index "scan_id_idx" {
```

Comment:
same here, since scan_id is autogenerated, maybe a unique assertion is not required here

Reply:
Double checked - the identity is not sufficient to provide a unique constraint. Furthermore, I tried adding a constraint via sql query and inspected to see what atlas would generate - it just produces the index as well so it would appear that is the only way to do it.